graphql_compiler/macros/macro_edge/helpers.py

```diff
@@ -140,6 +144,46 @@ def find_target_and_copy_path_to_it(ast):
         return new_ast, target_ast
 
 
+def _get_type_at_target(schema, ast, current_type):
+    """Return type at target or none if there is no target."""
+    # Base case
+    for directive in ast.directives:
+        if directive.name.value == MacroEdgeTargetDirective.name:
+            return current_type
+
+    # Recurse
+    if isinstance(ast, (Field, InlineFragment, OperationDefinition)):
+        if ast.selection_set is not None:
+            for selection in ast.selection_set.selections:
+                type_in_selection = None
+                if isinstance(selection, Field):
+                    if selection.selection_set is not None:
+                        type_in_selection = get_vertex_field_type(
+                            current_type, selection.name.value)
+                elif isinstance(selection, InlineFragment):
+                    type_in_selection = schema.get_type(selection.type_condition.name.value)
+                else:
+                    raise AssertionError(u'Unexpected selection type received: {} {}'
+                                         .format(type(selection), selection))
+
+                if type_in_selection is not None:
+                    type_at_target = _get_type_at_target(schema, selection, type_in_selection)
+                    if type_at_target is not None:
+                        return type_at_target
+
+    else:
+        raise AssertionError(u'Unexpected AST type received: {} {}'.format(type(ast), ast))
```

Comment:
Maybe switch the order here and do a `if not isinstance(...): raise AssertionError()`, for the sake of readability? Then you don't need to indent the meat of the function.